src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -1347,35 +1347,46 @@ internal static object CompareOperators(ExecutionContext context, IScriptExtent
         }
```

Comment:
@iSazonov, your last commit had 10 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

Split Operator.Binary split operator.Binary split operator can works with max substring limit
```powershell
Expected strings to be the same, but they were different.
Expected length: 5
Actual length:   1
Strings differ at index 1.
Expected: 'a b c'
But was:  'a'
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Language/Operators/SplitOperator.Tests.ps1: line 38
38:             $res[0] | Should -Be "a b c"
```
Split Operator.Binary split operator.Binary split operator can work with regex expression
```powershell
Expected strings to be the same, but they were different.
Expected length: 5
Actual length:   1
Strings differ at index 1.
Expected: 'a2b3c'
But was:  'a'
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Language/Operators/SplitOperator.Tests.ps1: line 106
106:             $res[0] | Should -Be "a2b3c"
```
Split Operator.Binary split operator options.Binary split operator works with IgnorePatternWhitespace
```powershell
Expected 3, but got 2.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Language/Operators/SplitOperator.Tests.ps1: line 245
245:             $res.count | Should -Be 3
```
Split Operator.Binary split operator options.Binary split operator works with ExplicitCapture
```powershell
Expected 2, but got 3.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Language/Operators/SplitOperator.Tests.ps1: line 259
259:             $res.count | Should -Be 2
```
Split Operator.Binary split operator options.Binary split operator works with Multiline (new line = '`n')
```powershell
Expected 2, but got 1.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Language/Operators/SplitOperator.Tests.ps1: line 212
212:             $res.count | Should -Be 2
```